### PR TITLE
Sagemaker

### PR DESCRIPTION
**Pull request by NeQuissimus**
###### Motivation for this change
Add sagemaker Python SDK

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NeQuissimus** [comment]
Packaging works for me :D

```
λ nix-shell -p python3 -p python3Packages.sagemaker -I nixpkgs=/opt/dev/upstream_nix/

[nix-shell:/opt/dev/upstream_nix]$ python3
Python 3.8.7 (default, Dec 21 2020, 17:18:55) 
[GCC 10.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from sagemaker.predictor import Predictor
pandas failed to import. Analytics features will be impaired or broken.
>>> from sagemaker.serializers import JSONSerializer
>>> predictor = Predictor("my-endpoint", serializer=JSONSerializer())
>>> predictor.predict({'command':'boo iexsjdkhlfhasdlkfjhakldhfaskdjhf'})
b'{\n    "outputs": [\n        [\n            0.0182475262\n        ]\n    ]\n}'
>>> 
```

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 113338` at 59cca617 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/c762072f1ccfcb9fab4d264a0022f76a">python37Packages.protobuf3-to-dict</a></li></li>
    <li><a href="https://gist.github.com/60ba3f2d7476254b50082979338e2ace">python37Packages.sagemaker</a></li></li>
    <li><a href="https://gist.github.com/0c940c027a2a3a84f9bd3f036512f053">python37Packages.smdebug-rulesconfig</a></li></li>
    <li><a href="https://gist.github.com/9bef0c8a627c72714999a8606ee02be6">python38Packages.protobuf3-to-dict</a></li></li>
    <li><a href="https://gist.github.com/9a6a73ecc293dd0dd48a194c3eff76a7">python38Packages.sagemaker</a></li></li>
    <li><a href="https://gist.github.com/ca2c095e2760b5cff496b539a08b9578">python38Packages.smdebug-rulesconfig</a></li></li>
    <li><a href="https://gist.github.com/9e2157952a2be4f0c42b4dec892a6358">python39Packages.protobuf3-to-dict</a></li></li>
    <li><a href="https://gist.github.com/3c96de96d52c538ad45b59cb2ea45264">python39Packages.sagemaker</a></li></li>
    <li><a href="https://gist.github.com/8143958a4232c2d1ebc49ec09cc8b9f1">python39Packages.smdebug-rulesconfig</a></li></li>
  </ul>
</details>

File: pkgs/development/python-modules/protobuf3-to-dict/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+buildPythonPackage rec {
+  pname = "protobuf3-to-dict";
+  version = "0.1.5";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0nibblvj3n20zvq6d73zalbjqjby0w8ji5mim7inhn7vb9dw4hhy";
+  };
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/protobuf3-to-dict/default.nix:12]:
```suggestion
  doCheck = false;
  pythonImportsCheck = [ "protobuf_to_dict" ];
```

File: pkgs/development/python-modules/protobuf3-to-dict/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  propagatedBuildInputs = [ protobuf six ];
+
+  meta = with lib; {
+    description =
+      "A teeny Python library for creating Python dicts from protocol buffers and the reverse";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/protobuf3-to-dict/default.nix:18]:
```suggestion
    description = "A teeny Python library for creating Python dicts from protocol buffers and the reverse";
```

File: pkgs/development/python-modules/sagemaker/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+buildPythonPackage rec {
+  pname = "sagemaker";
+  version = "2.24.5";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1j1a058ic00yxnf0cc364fzn82pacih5ffrh5s4dw1q4s3by4cvd";
+  };
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sagemaker/default.nix:12]:
```suggestion
  doCheck = false;
  pythonImportsCheck = [ "sagemaker" ];
```

File: pkgs/development/python-modules/sagemaker/default.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+  propagatedBuildInputs = [ attrs boto3 google-pasta importlib-metadata numpy protobuf protobuf3-to-dict smdebug-rulesconfig ];
+
+  meta = with lib; {
+    description =
+      "Library for training and deploying machine learning models on Amazon SageMaker";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sagemaker/default.nix:18]:
```suggestion
    description = "Library for training and deploying machine learning models on Amazon SageMaker";
```

File: pkgs/development/python-modules/smdebug-rulesconfig/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  pname = "smdebug-rulesconfig";
+  version = "1.0.1";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "smdebug_rulesconfig";
+    sha256 = "1mpwjfvpmryqqwlbyf500584jclgm3vnxa740yyfzkvb5vmyc6bs";
+  };
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/smdebug-rulesconfig/default.nix:13]:
```suggestion
  doCheck = false;
  pythonImportsCheck = [ "smdebug_rulesconfig" ];
```

File: pkgs/development/python-modules/smdebug-rulesconfig/default.nix [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+  propagatedBuildInputs = [  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/smdebug-rulesconfig/default.nix:15]:
```suggestion
```

File: pkgs/development/python-modules/smdebug-rulesconfig/default.nix [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+  meta = with lib; {
+    description =
+      "These builtin rules are available in Amazon SageMaker";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/smdebug-rulesconfig/default.nix:19]:
```suggestion
    description = "These builtin rules are available in Amazon SageMaker";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/sagemaker/default.nix [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  version = "2.24.5";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1j1a058ic00yxnf0cc364fzn82pacih5ffrh5s4dw1q4s3by4cvd";
+  };
+
+  doCheck = false;
+
+  propagatedBuildInputs = [ attrs boto3 google-pasta importlib-metadata numpy protobuf protobuf3-to-dict smdebug-rulesconfig ];
```
Comment (jonringer) [inline pkgs/development/python-modules/sagemaker/default.nix:14]: getting a little long
```suggestion
  propagatedBuildInputs = [
    attrs
    boto3
    google-pasta
    importlib-metadata
    numpy
    protobuf
    protobuf3-to-dict
    smdebug-rulesconfig
  ];
```

**Review by jonringer — COMMENTED** [review]
otherwise LGTM after @SuperSandro2000 's suggestions

File: pkgs/development/python-modules/protobuf3-to-dict/default.nix (hunk 2 above)
Comment (NeQuissimus) [reply pkgs/development/python-modules/protobuf3-to-dict/default.nix:18]: What do you use to format, @SuperSandro2000 ? `nixfmt` put this line break :)

Comment (SuperSandro2000) [reply pkgs/development/python-modules/protobuf3-to-dict/default.nix:20]: nixpkgs-fmt which does less line breaks like this which is annoying when doing treewide changes.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 113338` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>python37Packages.protobuf3-to-dict</li>
<li>python37Packages.smdebug-rulesconfig</li>
<li>python38Packages.protobuf3-to-dict</li>
<li>python38Packages.smdebug-rulesconfig</li>
<li>python39Packages.protobuf3-to-dict</li>
<li>python39Packages.smdebug-rulesconfig</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 113338` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>python37Packages.protobuf3-to-dict</li>
<li>python37Packages.smdebug-rulesconfig</li>
<li>python38Packages.protobuf3-to-dict</li>
<li>python38Packages.smdebug-rulesconfig</li>
<li>python39Packages.protobuf3-to-dict</li>
<li>python39Packages.smdebug-rulesconfig</li>
</ul>
</details>

**Pull request merged**